synapse/lib/stormtypes.py

```diff
@@ -8489,6 +8636,38 @@ def fromprim(valu, path=None, basetypes=True):
 
     return valu
 
+async def tocmprvalu(valu):
+
+    if isinstance(valu, (str, int, bool, float, bytes, types.AsyncGeneratorType, types.GeneratorType, Number)) or valu is None:
+        return valu
+
+    if isinstance(valu, (tuple, list)):
+        retn = []
+        for v in valu:
+            try:
+                retn.append(await toprim(v))
```

Comment:
Do these cases need to be tocmprvalu() recursion like they are in the toprim case?